pkg/build/strategies/sti/sti.go

```diff
@@ -273,7 +274,9 @@ func (builder *STI) Prepare(config *api.Config) error {
 
 	// fetch sources, for their .s2i/bin might contain s2i scripts
 	if len(config.Source) > 0 {
-		if builder.sourceInfo, err = builder.source.Download(config); err != nil {
+		// NOTE: needs timing
+		builder.sourceInfo, err = duration.DownloadTimer(builder.source.Download, builder.result.BuildInfo, config)
```

Comment:
this is not the right way to do this.  just note the start time, run the download, note the current time, subtract, done.  Creating a bunch of wrapper functions to do that isn't adding any value, you're not getting any reuse out of them.

but i don't want any of this duration related content in this PR.  your card for 3.4 is to get the failure reasons in place and i don't want this PR getting held up by discussion about how to handle the duration work which is for 3.5.